RevenueCatUI/CustomerCenter/Views/PromotionalOfferView.swift

```diff
@@ -26,6 +26,8 @@ import SwiftUI
 @available(visionOS, unavailable)
 struct PromotionalOfferView: View {
 
+    private let appearance: CustomerCenterConfigData.Appearance
```

Comment:
I'm wondering if it would also make sense to include this object in the environment so it's easier to access through the view tree, instead of having to pass it around... Wdyt?

Reply:
Would adding it to the environment not make it harder to test and/or preview? I have no evidence of that specific to SwiftUI haha, but that's generally a downside of implicit dependencies.

Reply:
It's a good point, however, I feel it's pretty easy to mock objects from the environment in SwiftUI for tests. If you make it an environment object, you would add it to the environment as a view modifier `.environmentObject(objectToAddToTheEnvironment)`.

When testing for snapshot tests, you can just inject a custom version of that object using the same modifier, instead of passing it as a parameter. Since I think we will likely only do screenshot tests on the views, I think that would be enough? Lmk what you think!

Reply:
I am thinking of including the whole configuration in the environment. What do you thnk?

Reply:
would you make a new `Appearance` class that conforms to `ObservableObject` and leave this struct? Or make the struct a class?

Reply:
Right, but you'd have to remember that you need to provide this mocked environment object, because there's nothing forcing you to do so (as opposed to e.g. constructor params). 

It's definitely not a blocker for me though! (I have yet to run into this 😛)

Reply:
yeah, it's kinda like mixing dependency injection and "singletons"

Reply:
> you'd have to remember that you need to provide this mocked environment object, because there's nothing forcing you to do so

That's true, it's the good thing about explicit dependencies... I do agree with you for dependencies that might only be used on specific places, but for "generic" objects like these (`Localization`, `Appearance`) that need to be accessed in a lot of places, I feel it's cleaner. Happy to discuss it more though!

> would you make a new Appearance class that conforms to ObservableObject and leave this struct? Or make the struct a class?

Hmm right.... Is it possible to pass it as a normal `environment` (not an EnvironmentObject)?

Reply:
Made it a `Environment`, it's quite easy to mock, and it cleans up the code a lot so leaving it like this for now